lib/github/filter.go

```diff
@@ -32,6 +33,44 @@ func (r Repository) IsPrivate() bool {
 	return r.Visibility == Private
 }
 
+// HasTopic indicates if a repository has a given topic.
+func (r Repository) HasTopic(topic string) bool {
+	for _, t := range r.Topics {
+		if t.Name == topic {
+			return true
+		}
+	}
+
+	return false
+}
+
+// UpdatedSince indicates if a repository has been updated since the given date.
+// Date has to be given in RFC3339 format, e.g. `2006-01-02T15:04:05Z07:00`.
+func (r Repository) UpdatedSince(date string) bool {
+	return timeSince(r.UpdatedAt, date)
+}
+
+// PushedSince indicates if a repository has been pushed since the given date.
+// Date has to be given in RFC3339 format, e.g. `2006-01-02T15:04:05Z07:00`.
+func (r Repository) PushedSince(date string) bool {
+	return timeSince(r.PushedAt, date)
+}
+
+// CreatedSince indicates if a repository has been created since the given date.
+// Date has to be given in RFC3339 format, e.g. `2006-01-02T15:04:05Z07:00`.
+func (r Repository) CreatedSince(date string) bool {
+	return timeSince(r.CreatedAt, date)
+}
+
+func timeSince(a time.Time, date string) bool {
```

Comment:
Consider renaming this method: timeSince sounds like it returns an amount of time, not a boolean. 
isAfter / isBefore?